lib/Onyx.js

```diff
@@ -1392,6 +1392,42 @@ function mergeCollection(collectionKey, collection) {
         });
 }
 
+/**
+ * Internal recursive function to execute the functions in the correct order
+ *
+ * @param {Array} data An array of objects with shape {onyxMethod: oneOf('set', 'merge', 'mergeCollection', 'multiSet', 'clear'), key: string, value: *}
+ * @returns {Promise<Void>}
+ */
+function innerUpdate(data) {
+    if (data.length === 0) {
+        return Promise.resolve();
+    }
+
+    const {onyxMethod, key, value} = data.shift();
+    let promise = Promise.resolve();
+    switch (onyxMethod) {
+        case METHOD.SET:
+            promise = set(key, value);
+            break;
+        case METHOD.MERGE:
+            promise = merge(key, value);
+            break;
+        case METHOD.MERGE_COLLECTION:
+            promise = mergeCollection(key, value);
+            break;
+        case METHOD.MULTI_SET:
+            promise = multiSet(value);
+            break;
+        case METHOD.CLEAR:
+            promise = clear();
+            break;
+        default:
+            break;
+    }
```

Comment:
It looks like this change possibly led to some regressions spotted in [this PR](https://github.com/Expensify/App/pull/29169#issuecomment-1755269426) and discussed in [this Slack thread](https://expensify.slack.com/archives/C035J5C9FAP/p1696941880028159?thread_ts=1696934662.824599&cid=C035J5C9FAP) that updated the Onyx version